### DISCUSSION TITLE
date/time localization and settings in 3.4.4

### DISCUSSION THREAD

**Discussion by tfboy** [netbox-community/netbox#12083] [Help Wanted!] [answered]
I recently updated my configuration.py file using the configuration_example.py file and editing the fields with my data (usernames, passwords, database, etc).

I do wonder if there is a bug though with the new `ENABLE_LOCALIZATION` parameter. Previously, this was set to `False`. If I set it to `True`, I lose all my local settings, despite having them defined at the bottom of my file.

Here's what I have:
```
# Localization
ENABLE_LOCALIZATION = True

# Time zone (default: UTC)
TIME_ZONE = 'CET'

# Date/time formatting. See the following link for supported formats:
# https://docs.djangoproject.com/en/stable/ref/templates/builtins/#date
DATE_FORMAT = 'j N Y'
SHORT_DATE_FORMAT = 'Y-m-d'
TIME_FORMAT = 'H:i'
SHORT_TIME_FORMAT = 'H:i:s'
DATETIME_FORMAT = 'j N Y H:i'
SHORT_DATETIME_FORMAT = 'Y-m-d H:i'
```

But I have a US date and 12h time format shown:
![image](https://user-images.githubusercontent.com/6972726/228497383-419974c4-cc6d-44e4-806a-debd1068b543.png)

I have `DEFAULT_LANGUAGE` set to `fr-FR` but this doesn't seem to have any impact.

Is this expected behaviour? My understanding was that if `ENABLE_LOCALISATION` was set to `True`, then the custom date/time settings at the end should be used, but it looks like they aren't.

**Comment by kkthxbye-code** [comment]
https://docs.netbox.dev/en/stable/configuration/system/#enable_localization

> Determines if localization features are enabled or not. This should only be enabled for **development or testing purposes** as netbox is not yet fully localized. Turning this on will localize numeric and date formats (**overriding what is set for DATE_FORMAT**) **based on the browser locale** as well as translate certain strings from third party modules.

It's working as described, it enables automatic localization based on the browser locale. Why do you want to enable it?

**Comment by candlerb** [comment, marked as answer]
https://docs.netbox.dev/en/stable/configuration/system/#enable_localization

_"Turning this on will localize numeric and date formats (overriding what is set for DATE_FORMAT) based on the browser locale"_

So as I understand it: it will intentionally ignore the date/time settings from configuration.py, and instead use whatever your browser says to use - as controlled by your browser's own localization settings.  So that's where you need to check next.

However, it's also not impossible that the feature has not been fully implemented.

I'm guessing this feature is following whatever the browser is setting in the `Accept-Language` header, so you can use tcpdump to see what your browser is sending:

```
tcpdump -i lo -nn -s0 -A tcp port 8001
```

**Reply by tfboy** [reply]
Thanks Brian. I was confused as my browser was set to English UK but still showing US units. Note to self: properly read the instructions!

**Reply by candlerb** [reply]
It *could* be a bug, but it could also be that it depends on the correct locale being installed/available in the server itself.

Is `en_GB` enabled in `/etc/locale.gen`? (If you edit this, you may need to run `locale-gen`).  But that's just a guess and I could well be wrong: it might be doing all the locale stuff in pure python.
